frontend/e-commerce/src/Components/Profile.js

```diff
@@ -0,0 +1,151 @@
+import { useEffect, useState } from "react";
+import { useSelector, useDispatch } from 'react-redux';
+import {selectProfile, loadProfile} from '../store/profileSlice';
+import FadeLoader from 'react-spinners/FadeLoader';
+import EditIcon from '@mui/icons-material/Edit';
+import {updateProfile} from '../Api';
+import { useNavigate, Link } from 'react-router-dom';
+import OrdersList from "./OrdersList";
+import styles from './Styles/Profile.css';
+
+
+function Profile() {
+    const profile = useSelector(selectProfile);
+    const { hasError, isLoading } = useSelector((state) => state.profile);
+    const dispatch = useDispatch();
+    const [showAddressForm, setShowAddressForm] = useState(false);
+    const [showPhoneForm, setShowPhoneForm] = useState(false);
+    const [Loading, setLoading] = useState(false);
+    const [address, setAddress] = useState(profile.address);
+    const [phone, setPhone] = useState(profile.phone);
+    const [fieldsFilled, setFieldsFilled] = useState(true);
+    const navigate = useNavigate();
+    const [showForm, setShowForm] = useState(false);
+
+    useEffect(() => {
+        dispatch(loadProfile());
```

Comment:
I know why it didn't work for you when you removed this line. 
This happens because after a login the `profile` is not updated inside the Redux store. 
To fix that you can do a `dispatch(loadProfile());` in the `<Login>` component just before you `navigate` to `/profile`.
This should allow you to remove this `useEffect`.